src/tls13.c

```diff
@@ -9268,9 +9334,28 @@ static int DoTls13NewSessionTicket(WOLFSSL* ssl, const byte* input,
     #ifdef WOLFSSL_EARLY_DATA
     ssl->session->maxEarlyDataSz  = ssl->options.maxEarlyDataSz;
     #endif
+
+#ifdef WOLFSSL_TICKET_NONCE_MALLOC
+    if (ssl->session->ticketNonce.data
+            != ssl->session->ticketNonce.dataStatic) {
+         XFREE(ssl->session->ticketNonce.data, ssl->heap,
+             DYNAMIC_TYPE_SESSION_TICK);
+         ssl->session->ticketNonce.data = ssl->session->ticketNonce.dataStatic;
+         ssl->session->ticketNonce.len = 0;
+    }
+    if (nonceLength > MAX_TICKET_NONCE_STATIC_SZ) {
+        WOLFSSL_MSG("Using dynamic nonce buffer");
+        ssl->session->ticketNonce.data = (byte*)XMALLOC(nonceLength,
+            ssl->heap, DYNAMIC_TYPE_SESSION_TICK);
+        if (ssl->session->ticketNonce.data == NULL)
+            return MEMORY_ERROR;
+    }
+#endif /* WOLFSSL_TICKET_NONCE_MALLOC */
```

Comment:
Almost this exact same sequence is found in `wolfSSL_d2i_SSL_SESSION()`. Maybe refactor into `PopulateSessionNonce()` type function?

Reply:
Done